jobs/build/signed-compose/build.groovy

```diff
@@ -4,96 +4,406 @@ commonlib = buildlib.commonlib
+// amount of time (10 minutes).
 def signedComposeStateQE() {
-    buildlib.elliott "${elliottOpts} change-state --state QE ${advisoryOpt}"
+    buildlib.elliott("${elliottOpts} change-state --state QE ${advisoryOpt}")
 }
 
+// Poll the 'signed' status of all builds attached to the advisory. We
+// wrap this in a retry() because some times the builds won't get
+// signed in a reasonable amount of time (10 minutes). To account for
+// that possible condition we frob the state between NEW_FILES and
+// QE. Re-entering the QE state triggers the signing mechanism again.
 def signedComposeRpmsSigned() {
-    buildlib.elliott "${elliottOpts} poll-signed ${adisoryOpt}"
+    def signed = false
+    retry(3) {
+        try {
+            buildlib.elliott("${elliottOpts} poll-signed --minutes=10 ${advisoryOpt}")
+            signed = true
+        } catch (err) {
+            signedComposeStateNewFiles()
+            // Give it a chance to catch up. ET can be slow
+            sleep 5
+            signedComposeStateQE()
+        } finally {
+            if ( !signed ) {
+                error("Failed to get all builds to 'signed' status after 3 retries")
+            }
+        }
+    }
 }
 
-def signedComposeNewCompose() {
-    echo "Puddle a signed compose including the errata"
+// Create a new RHEL7 compose. Take note of the following:
+//
+// The `puddle` command (which this step invokes) *REQUIRES* that the
+// user running it (ocp-build, in this case) has a valid kerberos
+// ticket. To ensure that this is ready and available for our use case
+// we must ensure a few things happen:
+//
+// * The `jenkins` user has enabled kerberos `ticket forwarding`
+//   option. When running `kinit` we *MUST* include the `-f` option
+//   flag. This requests a `forwardable` ticket for the user.
+//
+// * Additionally, ensure that the `jenkins` user has the:
+//
+//     `GSSAPIDelegateCredentials yes`
+//
+//  option set in their ~/.ssh/config file like this:
+//
+//     Host rcm-guest rcm-guest.app.eng.bos.redhat.com
+//     Hostname                   rcm-guest.app.eng.bos.redhat.com
+//     User                       ocp-build
+//     GSSAPIDelegateCredentials  yes
+//
+// That last line there, will tell the `ssh` command to try to
+// authenticate using GSSAPI (kerberos, in our case) credentials.
+//
+// FINALLY, the user on the remote end (`ocp-build`) *MUST* have their
+// `~/.k5login` file configured to include the principal we're
+// authenticating with.
+def signedComposeNewComposeEl7() {
+    if ( params.DRY_RUN ) {
+        currentBuild.description += "\nDry-run: EL7 Compose not actually built"
+        echo("Skipping running puddle. Would have used whitelist: ${errataList}")
+    } else {
+        def cmd = "ssh ocp-build@rcm-guest.app.eng.bos.redhat.com sh -s ${buildlib.args_to_string(params.BUILD_VERSION, errataList)} < ${env.WORKSPACE}/build-scripts/rcm-guest/call_puddle_advisory.sh"
```

Comment:
i should probably update `buildlib.invoke_on_rcm_guest` to use `commonlib.shell`
https://github.com/openshift/aos-cd-jobs/blob/94713e39041f7900b7478b7735015a60fdccff96/pipeline-scripts/buildlib.groovy#L658-L663

Reply:
more importantly, though, could we also use `ocp-tag-signed-errata-builds.py` for the el7 ones?